hyperspy/_components/polynomial.py

```diff
@@ -64,8 +64,11 @@ def grad_coefficients(self, x):
             return x * 0
 
     def __repr__(self):
-        return u'Component <%s order polynomial>'.encode('utf8') % (
-            ordinal(self.get_polynomial_order()))
+        text = u"%s order Polynomial component" % ordinal(
+            self.get_polynomial_order())
+        if self.name:
+            text = u"%s (%s)" % (self.name, text)
+        return u"<%s>".encode('utf8') % text
```

Comment:
I see, but could you provide an example where ``encode`` fixes
something?
On Thu, 2015-08-06 at 08:44 -0700, Tomas Ostasevicius wrote:
> In hyperspy/_components/polynomial.py:
> > @@ -64,8 +64,11 @@ def grad_coefficients(self, x):
> >              return x * 0
> >  
> >      def __repr__(self):
> > -        return u'Component <%s order polynomial>'.encode('utf8') %
> (
> > -            ordinal(self.get_polynomial_order()))
> > +        text = u"%s order Polynomial component" % ordinal(
> > +            self.get_polynomial_order())
> > +        if self.name:
> > +            text = u"%s (%s)" % (self.name, text)
> > +        return u"<%s>".encode('utf8') % text
> I'll remove the u, but regarding encode, please see here:
> #526
> —
> Reply to this email directly or view it on GitHub.